File: BIPServer/src/com/ccf/bip/biz/system/authorization/mapper/SysFunction.java
package com.ccf.bip.biz.system.authorization.mapper;

import java.io.Serializable;
import java.util.ArrayList;

public class SysFunction implements Serializable {
	private static final long serialVersionUID = 6564811311245561099L;

	private String functionId;

	private String key;
	
    private String functionName;

    private String parentId;

    private String functionType;

    private String serverName;

    private String url;

    private String image;

    private String tag;

    private Short seq;
    
    private String assemblyname;
    
    private Boolean showToolBar;
    
    private Boolean useHotKey;
    
    private String formType;
    
    private ArrayList<SysFunction> buttonList;

    public String getFunctionId() {
        return functionId;
    }

    public void setFunctionId(String functionId) {
        this.functionId = functionId == null ? null : functionId.trim();
    }

    public String getKey() {
		return key;
	}

	public void setKey(String key) {
		this.key = key;
	}

	public String getFunctionName() {
        return functionName;
    }

    public void setFunctionName(String functionName) {
        this.functionName = functionName == null ? null : functionName.trim();
    }

    public String getParentId() {
        return parentId;
    }

    public void setParentId(String parentId) {
        this.parentId = parentId == null ? null : parentId.trim();
    }

    public String getFunctionType() {
        return functionType;
    }

    public void setFunctionType(String functionType) {
        this.functionType = functionType == null ? null : functionType.trim();
    }

    public String getServerName() {
        return serverName;
    }

    public void setServerName(String serverName) {
        this.serverName = serverName == null ? null : serverName.trim();
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url == null ? null : url.trim();
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image == null ? null : image.trim();
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag == null ? null : tag.trim();
    }

    public Short getSeq() {
        return seq;
    }

    public void setSeq(Short seq) {
        this.seq = seq;
    } 
    
    public String getAssemblyname() {
		return assemblyname;
	}

	public void setAssemblyname(String assemblyname) {
		this.assemblyname = assemblyname;
	}

	public Boolean getShowToolBar() {
		return showToolBar;
	}

	public void setShowToolBar(Boolean showToolBar) {
		this.showToolBar = showToolBar;
	}

	public Boolean getUseHotKey() {
		return useHotKey;
	}

	public void setUseHotKey(Boolean useHotKey) {
		this.useHotKey = useHotKey;
	}

	public String getFormType() {
		return formType;
	}

	public void setFormType(String formType) {
		this.formType = formType;
	}
	
	public ArrayList<SysFunction> getButtonList() {
		return buttonList;
	}

	public void setButtonList(ArrayList<SysFunction> buttonList) {
		this.buttonList = buttonList;
	}

	@Override
    public String toString() {
		StringBuilder sb = new StringBuilder();
        sb.append(getClass().getSimpleName());
        sb.append(" [");
        sb.append("Hash = ").append(hashCode());
        sb.append(", functionId=").append(functionId);
        sb.append(", functionName=").append(functionName);
        sb.append(", parentId=").append(parentId);
        sb.append(", functionType=").append(functionType);
        sb.append(", serverName=").append(serverName);
        sb.append(", url=").append(url);
        sb.append(", image=").append(image);
        sb.append(", tag=").append(tag);
        sb.append(", seq=").append(seq);
        sb.append(", assemblyname=").append(assemblyname);
        sb.append(", showToolBar=").append(showToolBar);
        sb.append(", useHotKey=").append(useHotKey);
        sb.append(", formType=").append(formType);
        sb.append(", serialVersionUID=").append(serialVersionUID);
        sb.append("]");
        return sb.toString();
    }
}

File: BIPClient/BIPFramework/entity/SysFunction.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace com.ccf.bip.biz.system.authorization.mapper
{
    [Serializable]
    public class SysFunction
    {
        private string functionId;

        public string FunctionId
        {
            get { return functionId; }
            set { functionId = value; }
        }

        private string key;

        public string Key
        {
            get { return key; }
            set { key = value; }
        }

        private string functionName;

        public string FunctionName
        {
            get { return functionName; }
            set { functionName = value; }
        }

        private string parentId;

        public string ParentId
        {
            get { return parentId; }
            set { parentId = value; }
        }

        private string functionType;

        public string FunctionType
        {
            get { return functionType; }
            set { functionType = value; }
        }

        private string serverName;

        public string ServerName
        {
            get { return serverName; }
            set { serverName = value; }
        }

        private string url;

        public string Url
        {
            get { return url; }
            set { url = value; }
        }

        private string image;

        public string Image
        {
            get { return image; }
            set { image = value; }
        }

        private string tag;

        public string Tag
        {
            get { return tag; }
            set { tag = value; }
        }

        private short seq;

        public short Seq
        {
            get { return seq; }
            set { seq = value; }
        }

        private string assemblyname;

        public string Assemblyname
        {
            get { return assemblyname; }
            set { assemblyname = value; }
        }

        private bool showToolBar;

        public bool ShowToolBar
        {
            get { return showToolBar; }
            set { showToolBar = value; }
        }

        private bool useHotKey;

        public bool UseHotKey
        {
            get { return useHotKey; }
            set { useHotKey = value; }
        }

        private string formType;

        public string FormType
        {
            get { return formType; }
            set { formType = value; }
        }

        private ArrayList buttonList;

        public ArrayList ButtonList
        {
            get { return buttonList; }
            set { buttonList = value; }
        }
    }
}
